pylint/extensions/check_raise_docs.py

```diff
@@ -0,0 +1,265 @@
+    """Checker for raised exception documentation in Sphinx, Google or Numpy
+    style docstrings
+
+    * Check that all explicity raised exceptions in a function are documented
+    in the function docstring. Caught exceptions are ignored.
+
+    Activate this checker by adding the line::
+
+        load-plugins=pylint.extensions.check_raise_docs
+
+    to the ``MASTER`` section of your ``.pylintrc``.
+
+    :param linter: linter object
+    :type linter: :class:`pylint.lint.PyLinter`
+    """
+    __implements__ = IAstroidChecker
+
+    name = 'raise_checks'
+    msgs = {
+        'W9010': ('Raising a "%s" is not documented.',
+                  'missing-raise-doc',
+                  'Please document that this exception type is raised.'),
+    }
+
+    options = (('accept-no-raise-doc',
+                {'default': True, 'type' : 'yn', 'metavar' : '<y or n>',
+                 'help': 'Whether to accept totally missing raises'
+                         'documentation in a docstring of a function.'
+                }),
+              )
+
+    priority = -2
+
+    def __init__(self, linter=None):
+        BaseChecker.__init__(self, linter)
+
+    re_sphinx_raise_in_docstring = re.compile(r"""
+        :raises                  # Sphinx keyword
+        \s+                     # whitespace
+
+        (?:                     # type declaration
+        (\w+)
+        \s+
+        )?
+
+        (\w+)                   # Parameter name
+        \s*                     # whitespace
+        :                       # final colon
+        """, re.X | re.S)
+
+    re_google_raise_section = re.compile(r"""
+        ^([ ]*)   Raises \s*:   \s*?$ # Google parameter header
+        (  .* )                       # section
+        """, re.X | re.S | re.M)
+
+    re_google_raise_line = re.compile(r"""
+        \s*  (\w+) \s* :              # identifier
+        \s*  ( \w+ )?                 # beginning of optional description
+    """, re.X)
+
+    re_numpy_raise_section = re.compile(r"""
+        ^([ ]*)   Raises   \s*?$       # Numpy parameters header
+        \s*     [-=]+   \s*?$          # underline
+        (  .* )                        # section
+    """, re.X | re.S | re.M)
+
+    re_numpy_raise_line = re.compile(r"""
+        \s*  (\w+)                    # type declaration
+    """, re.X)
+
+    def visit_raise(self, node):
+        func_node = node.frame()
+        if not isinstance(func_node, astroid.FunctionDef):
+            return
+
+        excs = _possible_exc_types(node)
+        if not excs:
+            return
+
+        doc = func_node.doc
+        if doc is None:
+            if self.config.accept_no_raise_doc:
+                return
+        elif self.re_sphinx_raise_in_docstring.search(doc) is not None:
+            self._check_sphinx_doc(doc, excs, func_node)
+            return
+        else:
+            match, is_google, re_line = self._establish_section_type(doc)
+            if match is None:
```

Comment:
In this case, I think it's worth to just return, instead of emitting these messages. If we can't determine the documentation style used by the user, we shouldn't emit it at all. Of course, here you need to take in account if its a google or numpy style, but if it doesn't belong in this set, just return.